daemon/commit.go

```diff
@@ -20,7 +21,10 @@ type ContainerCommitConfig struct {
 // The image can optionally be tagged into a repository.
 func (daemon *Daemon) Commit(container *Container, c *ContainerCommitConfig) (*image.Image, error) {
 	if c.Pause && !container.isPaused() {
-		container.pause()
+		if err := container.pause(); err != nil && err != derr.ErrorCodeNotRunning.WithArgs(container.ID) && err != derr.ErrorCodeAlreadyPaused.WithArgs(container.ID) {
```

Comment:
I think there is no need to cast 'err', ErrorCodeNotRunning is just a int value. We can just checck err's Code is ErrorCodeNotRunning or not